dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScopeTest.kt

```diff
@@ -111,828 +88,562 @@ internal class RumSessionScopeTest {
     lateinit var mockSessionListener: RumSessionListener
 
     @Mock
-    lateinit var mockBuildSdkVersionProvider: BuildSdkVersionProvider
+    lateinit var mockRumEventSourceProvider: RumEventSourceProvider
 
     @Mock
-    lateinit var mockRumEventSourceProvider: RumEventSourceProvider
+    lateinit var mockBuildSdkVersionProvider: BuildSdkVersionProvider
 
     @Forgery
     lateinit var fakeParentContext: RumContext
 
-    @Forgery
-    lateinit var fakeUserInfo: UserInfo
-
-    @Forgery
-    lateinit var fakeNetworkInfo: NetworkInfo
-
     @FloatForgery(min = 0f, max = 100f)
     var fakeSamplingRate: Float = 0f
 
     @BoolForgery
     var fakeBackgroundTrackingEnabled: Boolean = false
 
+    lateinit var fakeInitialViewEvent: RumRawEvent
+
     @BeforeEach
-    fun `set up`() {
-        whenever(coreFeature.mockUserInfoProvider.getUserInfo()) doReturn fakeUserInfo
-        whenever(coreFeature.mockNetworkInfoProvider.getLatestNetworkInfo())
-            .thenReturn(fakeNetworkInfo)
+    fun `set up`(forge: Forge) {
+        fakeInitialViewEvent = forge.startViewEvent()
+
         whenever(mockParentScope.getRumContext()) doReturn fakeParentContext
         whenever(mockChildScope.handleEvent(any(), any())) doReturn mockChildScope
-        whenever(mockChildScope.isActive()) doReturn true
-        whenever(mockBuildSdkVersionProvider.version()) doReturn Build.VERSION_CODES.BASE
 
-        CoreFeature.processImportance = RunningAppProcessInfo.IMPORTANCE_FOREGROUND
+        initializeTestedScope()
+    }
 
-        testedScope = RumSessionScope(
-            mockParentScope,
-            100f,
-            fakeBackgroundTrackingEnabled,
-            mockDetector,
-            mockCpuVitalMonitor,
-            mockMemoryVitalMonitor,
-            mockFrameRateVitalMonitor,
-            mockTimeProvider,
-            mockSessionListener,
-            mockRumEventSourceProvider,
-            mockBuildSdkVersionProvider,
-            TEST_INACTIVITY_NS,
-            TEST_MAX_DURATION_NS
-        )
-        val originalRumContext = testedScope.getRumContext()
-        assertThat(GlobalRum.getRumContext()).isEqualTo(originalRumContext)
+    @AfterEach
+    fun `tear down`() {
+        // whatever happens
+        assertThat(testedScope.isActive()).isTrue()
```

Comment:
why this one is needed if `RumSessionScope#isActive` always returns `true`?

Reply:
Because we want to test that it always returns true.